tensorhive/core/utils/reservation_allowed.py

```diff
@@ -0,0 +1,78 @@
+from datetime import time, timedelta
```

Comment:
`reservation_allowed` does not seem like a good name for a file... it doesn't really tell me what this file would do.

My suggestion would be to call it, for example `ReservationVerifier.py` and in here create a class `ReservationVerifier` that would have all of these methods below.